src/Nethermind/Nethermind.Db/DbTracker.cs

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using NonBlocking;
+
+namespace Nethermind.Db;
+
+public class DbTracker
+{
+    private readonly ConcurrentDictionary<string, IDbMeta> _createdDbs = new ConcurrentDictionary<string, IDbMeta>();
+
+    public void AddDb(string name, IDbMeta dbMeta)
+    {
+        _createdDbs.TryAdd(name, dbMeta);
```

Comment:
Can this mean the db gets lost; as you add but fail because it already exists; however return one passed in?

Should it be GetOrAdd?

Does the one not added need to be disposed?